massa-hash/src/hash.rs

```diff
@@ -74,30 +74,16 @@ impl std::fmt::Debug for Hash {
     }
 }
 
-/// The bitwise XOR is important to maintain the latest hash of the ~ 1TB ledger. Since the size is big,
-/// it is not feasible to hash the whole ledger every 0.5s (when the changes occur). Instead,
-/// all the rows of the ledger are hashed and XORed together to get a single hash. Whenever, any
-/// row has to be changed, the row's original hash is XORed with the single hash. This essentially nullifies
-/// its effect and then the new hash of the changed row is XORed with the single hash. Thus,
-/// giving the latest hash without going over the entire ~1TB ledger. This method is as secure as the traditional
-/// hash at the same time, it is much more efficient and convenient when the throughput needs are high and the requirement
-/// of integrity cannot be compromised.
-
+/// We use bitwise XOR for hashes on some structures.
+/// However, we no longer use this for the final state hash, as we use lsmtree's Sparse Merkle Tree instead.
 impl BitXorAssign for Hash {
     fn bitxor_assign(&mut self, rhs: Self) {
         *self = *self ^ rhs;
     }
 }
 
-/// The bitwise XOR is important to maintain the latest hash of the ~ 1TB ledger. Since the size is big,
-/// it is not feasible to hash the whole ledger every 0.5s (when the changes occur). Instead,
-/// all the rows of the ledger are hashed and XORed together to get a single hash. Whenever, any
-/// row has to be changed, the row's original hash is XORed with the single hash. This essentially nullifies
-/// its effect and then the new hash of the changed row is XORed with the single hash. Thus,
-/// giving the latest hash without going over the entire ~1TB ledger. This method is as secure as the traditional
-/// hash at the same time, it is much more efficient and convenient when the throughput needs are high and the requirement
-/// of integrity cannot be compromised.
-
+/// We use bitwise XOR for hashes on some structures.
```

Comment:
Previous comment was also duplicated, don't know if it's necessary or not (in case the doc_comment is displayed somewhere else by the IDE for instance).